app/components/primer/auto_complete_component.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use AutoComplete to populate input values from server search results.
+  class AutoCompleteComponent < Primer::Component
+    include ViewComponent::SlotableV2
+
+    DEFAULT_INPUT_TYPE = :text
+    INPUT_TYPE_OPTIONS = [DEFAULT_INPUT_TYPE, :search].freeze
+
+    # Required input used to search for results
+    #
+    # @param type [Symbol] <%= one_of(Primer::AutoCompleteComponent::INPUT_TYPE_OPTIONS) %>
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :input, lambda { |type: DEFAULT_INPUT_TYPE, **system_arguments|
+      system_arguments[:tag] = :input
+      system_arguments[:type] = fetch_or_fallback(INPUT_TYPE_OPTIONS, type, DEFAULT_INPUT_TYPE)
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # Optional icon to be rendered before the input. Has the same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
+    renders_one :icon, Primer::OcticonComponent
+
+    # Customizable results list.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :results, lambda { |**system_arguments|
+      system_arguments[:tag] = :ul
+      system_arguments[:id] = @id
+      system_arguments[:classes] = class_names(
+        "autocomplete-results",
+        system_arguments[:classes]
+      )
+
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # @example Default
+    #   <%= render(Primer::AutoCompleteComponent.new(src: "/users/search", id: "user-popup", position: :relative)) do |c| %>
+    #     <% c.input(type: :text, name: "input", classes: "form-control") %>
+    #     <% c.results do %>
```

Comment:
I wonder what you'd think about a single `result` being a slot so the api would be `c.result`. I suppose users may want to style the `results` container so we'd want that to be a slot. Thinking out loud here. WDYT?

Reply:
the issue is that the results are populated async, so what would actually return each result is the `src` you are calling